lib/api/controllers/securityController.js

```diff
@@ -423,20 +465,7 @@ class SecurityController {
    * @returns {Promise<Object>}
    */
   createUser(request) {
-    const user = new User();
-    let pojoUser;
-
-    assertHasBody(request);
-    assertBodyHasAttribute(request, 'profileIds');
-    assertBodyAttributeType(request, 'profileIds', 'array');
-    assertIdStartsNotUnderscore(request);
-
-    pojoUser = request.input.body;
-    pojoUser._id = request.input.resource._id || uuid.v4();
-
-    return this.kuzzle.repositories.user.hydrate(user, pojoUser)
-      .then(modifiedUser => this.kuzzle.repositories.user.persist(modifiedUser, { database: {method: 'create'} }))
-      .then(modifiedUser => formatProcessing.formatUserForSerialization(this.kuzzle, modifiedUser));
+    return createUser(this.kuzzle, request, false);
```

Comment:
nitpicking: could keep the old code with the only changes needed by the new user format (see also comment below):

```javascript
createUser(request) {
  assertHasBody(request);
  assertIdStartsNotUnderscore(request);
  assertBodyHasAttribute(request, 'content');
  assertBodyAttributeType(request, 'content', 'object');

  // new assertions methods to create:
  assertContentHasAttribute(request.input.body.content, 'profileIds');
  assertContentAttributeType(request.input.body.content, 'profileIds', 'object');

  const
    user = new User(),
    pojoUser = request.input.body;

  pojoUser._id = request.input.resource._id || uuid.v4();

  return Bluebird.coroutine(persistUserAndCredentials)(kuzzle, request, pojoUser);
}
```